pkg/auth/providers/activedirectory/activedirectory_client.go

```diff
@@ -323,8 +323,13 @@ func (p *adProvider) searchPrincipals(name, principalType string, config *v3.Act
 }
 
 func (p *adProvider) searchUser(name string, config *v3.ActiveDirectoryConfig, caPool *x509.CertPool) ([]v3.Principal, error) {
-	query := "(&(" + config.UserSearchAttribute + "=*" + name + "*)(" + ObjectClassAttribute + "=" +
-		config.UserObjectClass + "))"
+	srchAttributes := strings.Split(config.UserSearchAttribute, "|")
+	query := "(&(" + ObjectClassAttribute + "=" + config.UserObjectClass + ")"
+	srchAttrs := "(|"
```

Comment:
What will happen if `srchAttributes` is empty? this search wont function obviously, but i just want to make sure it fails gracefully

Reply:
@cjellick It fails gracefully, UI displays 'no options available' in that case

Reply:
Should we still make it a compulsory field?